### PR TITLE
Refactor(input model): create the input

### PR DESCRIPTION
**Pull request by grysgreat**
rebuild class and function. This has no effect on the current code.
 Refactor #2549

### CODE REVIEW

File: source/module_io/parameter_vector.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+#include <vector>
+
+template <typename T>
+class SimpleVector
```
Comment (dyzheng) [inline source/module_io/parameter_vector.h:15]: Please add annotations with doxygen format.

**Review by dyzheng — CHANGES_REQUESTED** [review]

File: source/module_io/test/parameter_pool_test.cpp [hunk 2]
```diff
@@ -0,0 +1,396 @@
+// #include "gtest/gtest.h"
```
Comment (dyzheng) [inline source/module_io/test/parameter_pool_test.cpp:1]: What is the usage of this file?

File: source/module_io/test/parameter_pool_set_test.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+#include <utility>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "module_io/parameter_pool.h"
+
+/**
+ * @brief test func:input_parameters_get() and input_parameters_set() in parameter_pool.h
+ */
+TEST(InputParameterArgsTest, InputTest)
```
Comment (dyzheng) [inline source/module_io/test/parameter_pool_set_test.cpp:19]: Do you expected other users use input module follow this UT? 
In this example, there is a risk usage of pointer cast:
`
inputparaBool.set((void*)&b1);
ModuleIO::input_parameters["bool"] = inputparaBool;
bool b2 = *static_cast<bool*>(ModuleIO::input_parameters["bool"].get());
`
A bool* -> void* ->bool* , if user misuse it as bool* ->void* ->double*, any error will be throwed?

**Review by dyzheng — CHANGES_REQUESTED** [review]

File: source/module_io/test/parameter_pool_test.cpp (hunk 2 above)
Comment (grysgreat) [reply source/module_io/test/parameter_pool_test.cpp:1]: This test code is for the formal access test of the module in the next stage. It cannot be used before the new input module is perfected, so it is not useful for the time being.

File: source/module_io/test/parameter_pool_set_test.cpp (hunk 3 above)
Comment (grysgreat) [reply source/module_io/test/parameter_pool_set_test.cpp:19]: This test code is used to test the correctness of parameter_input module parameter passing. For users, these codes do not need to be written, and the code logic has been written in the current module.
  parameter_input acts as a bridge between input files and input.h. It configures customizable configuration files (including default values and user-input values) to input.h. A map with parameters value and types is then generated -- ModuleIO::input_parameters["sth"]. In use (new requirements: printing all variables, for example) it is only necessary to know the correct type of sth and use the corresponding pointer.

Comment (grysgreat) [reply source/module_io/test/parameter_pool_set_test.cpp:19]: This test code is used to test the correctness of parameter_input module parameter passing. For users, these codes do not need to be written, and the code logic has been written in the current module.
parameter_input acts as a bridge between input files and input.h. It configures customizable configuration files (including default values and user-input values) to input.h. A map with parameters value and types is then generated -- ModuleIO::input_parameters["sth"]. In use (new requirements: printing all variables, for example) it is only necessary to know the correct type of sth and use the corresponding pointer.

**Review by dyzheng — APPROVED** [review]
I have no question for this PR.

**Review by grysgreat — COMMENTED** [review]
I'm no question to this change

**Comment by hongriTianqi** [comment]
Close as has been solved.

**Pull request merged**